src/kudu/consensus/leader_election.cc

```diff
@@ -1184,7 +1198,8 @@ LeaderElection::LeaderElection(RaftConfigPB config,
       timeout_(timeout),
       decision_callback_(std::move(decision_callback)),
       highest_voter_term_(0),
-      start_time_(MonoTime::Now()) {
+      start_time_(MonoTime::Now()),
+      vote_logger_(vote_logger) {
```

Comment:
```suggestion
      vote_logger_(std::move(vote_logger)) {
```

Prevents copying shared_ptr which needs to bump a ref count